### DISCUSSION TITLE
Automate Scripts to Update Custom Fields

### DISCUSSION THREAD

**Discussion by ben-marquardt-tfs** [netbox-community/netbox#9462] [Q&A] [answered]
I have some scripts that are used to update some of the custom fields for Sites/Providers/etc. The input for these scripts depends on the data associated with each Circuit. Is there a way to trigger a script to run when a circuit is updated or a new one is created? It would be really nice to have each field automatically update without having to manually run a script.

I assume it is possible but perhaps it is not. My first thought was that webhooks could do this but I am new to this and have never used webhooks. Any ideas?

**Comment by matejv** [comment, marked as answer]
Sounds like you should be able to use [custom validators](https://docs.netbox.dev/en/stable/customization/custom-validation/) for this.

Attach a validator to a circuit and you'll be able to run your code to modify providers... as you like. Be warned, that might be abusing the custom validation functionality a bit, though :)

**Reply by ben-marquardt-tfs** [reply]
Thanks for the reply! Im curious if you mentioned custom validators instead of webhooks because webhooks would not work in this case? 

Additionally, is this just an abuse of validation functionality because the custom validation is intended to ensure data integrity and not modify the data itself?

**Reply by candlerb** [reply]
Webhooks call external applications via HTTP.  So you would have to write a separate server, which accepts the webhook trigger, and then calls back to the Netbox API to fetch and/or update the necessary information.

Whether it's acceptable practice to perform database modifications inside a custom validator, I have no idea.

**Reply by matejv** [reply]
While you could configure a webhook to call a URL that would trigger running your script, I'm not sure if the webhook payload would be in a valid format.

There might be a third approach that might be the most "correct" way of doing this. Ther is a management commad to run a script form shell: [runscript](https://github.com/netbox-community/netbox/blob/develop/netbox/extras/management/commands/runscript.py). You could put this in cron, and run a scipt that loops your circuits and does what needs to be done.

**Reply by ben-marquardt-tfs** [reply]
Gotcha. And I was just looking at that, from what I know it seems like the simplest approach to do this. Thanks for the help!
